src/cached.rs

```diff
@@ -0,0 +1,73 @@
+use crate::Uncertain;
+use rand::Rng;
+use std::cell::Cell;
+
+/// Cached uncertainty. Its reference also implements Uncertain.
+/// See [`Uncertain::into_cached`].
+pub struct CachedUncertain<U>
+where
+    U: Uncertain,
+    U::Value: Clone,
+{
+    ptr: U,
```

Comment:
naming nit: this is not actually a pointer 😅
```suggestion
    uncertain: U,
```

Reply:
Sounds much better, thanks